### PR TITLE
feat(hackathon-surplus): add new surplus modal when order filled

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Continuing hackathon surplus project

Follow up to https://github.com/cowprotocol/cowswap/pull/2751

Display the new surplus modal when the order is filled

As you can see, it's very close to the design, only some minor stylings missing

|This|Design|
|-|-|
| Screenshot 2023-06-29 at 16 58 56 | image |

# ⚠️ Notes ⚠️
- Styling not included
- Share button has no data
- Link to `learn more` is missing
- Modal must be open after order placement to see it

# To Test

1. Place order
2. Keep order confirmation modal open
* Once order is filled, the new modal should be displayed

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @alfetopito , great, great modal! 
I noticed that the modal does not appear when the progress bar is closed. Should it be implemented later? :)

**Comment by alfetopito** [comment]
> Hey @alfetopito , great, great modal! I noticed that the modal does not appear when the progress bar is closed. Should it be implemented later? :)

Yes, this will be a follow up PR

**Review by elena-zh — APPROVED** [review]

File: src/legacy/components/TransactionConfirmationModal/index.tsx [hunk 1]
```diff
@@ -70,8 +73,18 @@ export function TransactionConfirmationModal({
...
 }
+
+function getWidth(hash: string | undefined, activityDerivedState: ActivityDerivedState | null): number {
+  if (activityDerivedState && getActivityState(activityDerivedState) === 'filled') {
+    return 400
+  }
+  if (hash) {
+    return 850
+  }
+  return 470
```
Comment (anxolin) [inline src/legacy/components/TransactionConfirmationModal/index.tsx:89]: im curious what are these 2 cases, the 850 and 470
maybe a comment giving them a name, or a named const would help

**Review by anxolin — COMMENTED** [review]
approve

**Pull request merged**